src/ai/backend/common/types.py

```diff
@@ -476,6 +476,29 @@ class MovingStatValue(TypedDict):
     },
 )
 
+MetricValueFieldKey = NewType("MetricValueFieldKey", str)
+MetricValueFieldPair = tuple[MetricValueFieldKey, Optional[str]]
+
+CURRENT_METRIC_KEY = MetricValueFieldKey("current")
+CAPACITY_METRIC_KEY = MetricValueFieldKey("capacity")
+PCT_METRIC_KEY = MetricValueFieldKey("pct")
+
+
+@dataclass
+class FlattenedKernelMetric:
+    agent_id: AgentId
+    kernel_id: KernelId
+    key: MetricKey
+    value_pairs: list[MetricValueFieldPair]
+
+
+@dataclass
+class FlattenedDeviceMetric:
+    agent_id: AgentId
+    device_id: DeviceId
+    key: MetricKey
+    value_pairs: list[MetricValueFieldPair]
```

Comment:
It's a little strange that the metric is in common/types.py, is there a reason you put it there?